portal-impl/test/unit/com/liferay/portal/util/FileImplTest.java

```diff
@@ -223,6 +236,134 @@ public void testStripSuffixWhenFileNameHasParenthesisAtStart() {
 			"()test.jsp", _fileImpl.stripParentheticalSuffix("()test.jsp"));
 	}
 
+	@Test
+	public void testUnzip() throws Exception {
+		_setUpForUnzipTests();
+
+		File file = _createZipFile(
+			"test.zip",
+			HashMapBuilder.put(
+				"zip/test/directory/", () -> (String)null
+			).put(
+				"zip/test/entry/entry.txt", "Test String"
+			).build());
+
+		_fileImpl.unzip(file, _tempFolder);
+
+		_assertExists("zip/test/directory");
+		_assertExists("zip/test/entry/entry.txt");
+
+		Assert.assertTrue(
+			_canFileReadWrite(_getTempTestFilePath("zip/test/directory")));
+
+		Assert.assertTrue(
+			_canFileReadWrite(
+				_getTempTestFilePath("zip/test/entry/entry.txt")));
+		_tearDown();
```

Comment:
@jeffwu0724 When any of the operation or assertion fails, the tear down method won't be executed. It should at least be protected using `try-finally` or `file.deleteOnExit()`.